src/test/java/com/google/api/codegen/testdata/php_library.baseline

```diff
@@ -145,10 +121,53 @@ class LibraryServiceApi
      */
     public static function parseBookFromBookName($bookName)
     {
-        return self::$bookNameTemplate->match($bookName)['book'];
+        return self::getBookNameTemplate()->match($bookName)['book'];
+    }
+
+
+    private static function getShelfNameTemplate()
+    {
+        if (self::$shelfNameTemplate == null) {
+            self::$shelfNameTemplate = new PathTemplate("shelves/{shelf}");
+        }
+        return self::$shelfNameTemplate;
+    }
+
+    private static function getBookNameTemplate()
+    {
+        if (self::$bookNameTemplate == null) {
+            self::$bookNameTemplate = new PathTemplate("shelves/{shelf}/books/{book}");
+        }
+        return self::$bookNameTemplate;
     }
 
 
+    private static function getPageStreamingDescriptors()
+    {
+        $listShelvesPageStreamingDescriptor =
+                new PageStreamingDescriptor([
+                    'requestPageTokenField' => 'page_token',
+                    'responsePageTokenField' => 'next_page_token',
+                    'resourceField' => 'shelves']);
+        $listBooksPageStreamingDescriptor =
+                new PageStreamingDescriptor([
+                    'requestPageTokenField' => 'page_token',
+                    'responsePageTokenField' => 'next_page_token',
+                    'resourceField' => 'books']);
+        $listStringsPageStreamingDescriptor =
+                new PageStreamingDescriptor([
+                    'requestPageTokenField' => 'page_token',
+                    'responsePageTokenField' => 'next_page_token',
+                    'resourceField' => 'strings']);
+
+        $pageStreamingDescriptors = [
```

Comment:
I am not concerned about the cost but more about the potential risks of issues. For example, if we need to modify this $pageStreamingDescriptors and call getPageStreamingDescriptors(), the modification will not be preserved. (However since it is a static get method, I expect the modifications are preserved)
It is okay as-is here since pageStreamingDescriptors will not be modified.